frontend/src/utils/ScheduleChangeTracker.ts

```diff
@@ -0,0 +1,71 @@
+import { convertTermIdToSeasonString, convertTermIdToYear } from ".";
+
+/**
+ * The ScheduleChangeTracker class defines the `getInstance` method that lets clients access
+ * the unique ScheduleChangeTracker instance.
+ */
+class ScheduleChangeTracker {
```

Comment:
I think this is ok for the time being. Other solutions I can think of are to create a new redux slice, or make use of React contexts. I think either of those would be "better practice", but since you already have this and it's working _and we're on a time crunch_ I don't see why we can go for it. 

One thing that we should test is that everything works as expected if an advisor goes into a student, makes some changes, then goes into a different student and makes changes.